core/src/main/scala/org/apache/spark/TaskContextImpl.scala

```diff
@@ -82,9 +97,27 @@ private[spark] class TaskContextImpl(
     }
   }
 
-  /** Marks the task for interruption, i.e. cancellation. */
+  /**
+   * Marks the task as interruption, i.e. cancellation. We add this
```

Comment:
I'd reword this comment slightly to say something like "Marks the task for interruption (i.e. cancellation) and calls any task interruption callbacks registered for this task."

We already document the sample use-case on the public addTaskInterruptionListener callback, so I don't think we need to do it here as well.